metals/src/main/scala/scala/meta/internal/metals/testProvider/BuildTargetUpdate.scala

```diff
@@ -0,0 +1,78 @@
+package scala.meta.internal.metals.testProvider
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.testProvider.TestExplorerEvent._
+import scala.meta.internal.mtags
+
+import ch.epfl.scala.bsp4j.BuildTarget
+import org.eclipse.{lsp4j => l}
+final case class BuildTargetUpdate(
+    targetName: String,
+    targetUri: String,
+    events: java.util.List[TestExplorerEvent]
+)
+object BuildTargetUpdate {
+  def apply(
+      buildTarget: BuildTarget,
+      events: Seq[TestExplorerEvent]
+  ): BuildTargetUpdate =
+    BuildTargetUpdate(
+      buildTarget.getDisplayName,
+      buildTarget.getId.getUri,
+      events.asJava
+    )
+}
+
+sealed abstract class TestExplorerEvent(
+    val kind: String,
+    val fullyQualifiedClassName: String,
+    val className: String
+) {
+  def fullyQualifiedName: FullyQualifiedName
+  def clsName: ClassName
+}
+object TestExplorerEvent {
+
+  final case class FullyQualifiedName(value: String) extends AnyVal
+  final case class ClassName(value: String) extends AnyVal
+  final case class RemoveTestSuite(
+      @transient fullyQualifiedName: FullyQualifiedName,
+      @transient clsName: ClassName
```

Comment:
Why do we create the class, mark them with transient and then do `.value` on them? Wouldn't it be simpler to avoid inheritance at all?

Reply:
`FullyQualifiedName` and `ClassName` are value classes because working on bare strings is very errorprone (for the record yes, I made a few mistakes before these value classes).

`@transient` is a leftover, now it can be removed, thanks!